python_ta/transforms/type_inference_visitor.py

```diff
@@ -258,11 +258,21 @@ def register_type_constraints_setter():
     return type_visitor
 
 
+def _populate_type_env(node):
+    """Helper to prevent overlap in populating locals and globals attributes in type environment of given node."""
+    if isinstance(node, astroid.FunctionDef):
+        node.type_environment = Environment(locals_={name: TYPE_CONSTRAINTS.fresh_tvar() for name in node.locals})
+    node.type_environment = Environment(globals_={name: TYPE_CONSTRAINTS.fresh_tvar() for name in node.globals})
```

Comment:
This does not actually fix the problem mentioned earlier. The implementations of the `_set_environment` method are different enough for the two classes (`Module` and `FunctionDef`) that you should just split them up into two entirely different functions.